src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -287,27 +287,35 @@
    :value    comment})
 
 (defn- create-authority-key-identifier
-  [public-key issuer-dn serial critical]
+  [public-key issuer-dn serial cert critical]
```

Comment:
I kept the same method name to minimize the amount of code I would have to change in puppetserver to use this new functionality. I would argue that the `SSLExtension` schema leaves the `:value` intentionally vague as merely `Object` just for this sort of use case. That brings up the 2nd sort of counter-argument, is that renaming it doesn't change the fact that it must conform to the `SSLExtension` schema, so we still end up blurring that difference between a representation of an auth-key-id and a bundle of options.

I could rename the function, but I kinda thought it was public api, since puppetserver calls it directly...so I could alias it to a new name?

Reply:
What if there was an additional `options` key added to `SSLExtensions` that carried the `truncated` and other non-oid specific data that we need to have in the objectmap for `parseExtensionObject`? Is that preferable to stuffing more data into `:values`?